### ISSUE TITLE
Accuracy of `jacpts` for very large n

### ISSUE THREAD

**Issue by ajt60gaibb** [chebfun/chebfun#1707]
James Bremer, Associate Professor at UC Davis, just told me about the following inaccuracies in `jacpts`: 

```
[x, w] =jacpts(10^6, -1/2,0);
sum(w) - 2*sqrt(2)
ans =
  -5.2684e-09
```

I do not think this is rounding error due to the `sum` command. More likely, `jacpts` should have done one extra Newton iteration.

**Comment by nickhale** [comment]
I think this is more complicated than simply requiring an additional Newton iteration. My guess is that the problem lies in the computing the final `cos(alpha)` term in the interior formula (which needs to be done accurately). It doesn't look like the faked (Hi-Lo) extended precision computation is working in some cases.

A more convenient example for verifying this is
```
>> [x, w] = jacpts(114, -1/2,0); sum(w) - 2*sqrt(2), k = k+1;
ans =
    -1.603162047558726e-13
```

**Comment by nickhale** [comment]
I think this is more complicated than simply requiring an additional iteration. Rather, I suspect the problem is in computing the final `cos(alpha)` values in the interior formulae, which needs to be done very accurately. We do this by faking an extended precision computation with a high-low factorisation, but it seems maybe this is failing in some cases.

A more convenient example for experimenting with is
```
>> [x, w] = jacpts(114, -1/2,0); sum(w) - 2*sqrt(2), k = k+1;
ans =
    -1.603162047558726e-13
```

**Comment by LourencoPeixoto** [comment]
I agree with Nick. The question is not to require an extra iteration. However I'm not sure that the problem is on computation of `cos(alpha)`. I'll investigate it.

**Comment by nickhale** [comment]
Closed by 95d2739 in #1702.

```
>> [x, w] = jacpts(n, a, b); sum(w) - 2*sqrt(2)
ans =
     5.329070518200751e-15
```

... this took me all morning to find. =(

**Comment by ajt60gaibb** [comment]
Sweet!

**Comment by nickhale** [comment]
Closed by #1702.

**Comment by LourencoPeixoto** [comment]
Great!

But the problem still remains in some cases. Check this example:
```
>> [x, w] = jacpts(1e5, 1,1);
>> sum(w)

ans =

   1.333333333031809
```
The exact sum is ```1.333333333333333```.

I think the best way to fix the constant for the weights is to use ratio of gamma functions (just like you presented at paper [2]). I've implemented a function on JACPTS to compute this ratio. The weights become much better:
```
>> [x, w] = jacpts(1e5, 1,1);
>> sum(w)

ans =

   1.333333333333337
```
I'm going create a pull request.

**Comment by nickhale** [comment]
Alternatively, couldn't we just do
```
C = 2^(a+b+1)*beta(a+1, b+1)/sum(w);
```
?

I'm torn between the two. The approach above is obviously simple, but going back the 'ratio of gamma functions' approach as suggested by @LourencoPeixoto has the advantage that each of the weights are still computed independently.

**Comment by LourencoPeixoto** [comment]
Yes, we could. In fact is simpler. On other hand the 'ratio of gamma functions' give slightly better results. Check the following with ```C = 2^(a+b+1)*beta(a+1, b+1)/sum(w)```:
```
[x, w] = jacpts(1e5, 1,1);
>> sum(w)

ans =

   1.333333333333339
``` 
Actually we can consider both with the same accuracy. However I had forgotten that the 'rec' method uses this constant too. And the 'ratio of gamma functions'  is inappropriate when ```n``` is small. For instance:
```
>> [x, w] = jacpts(5, 1,1);
>> sum(w)

ans =

   1.333333333305571
```
whereas the alternative way (beta/sum) suggested by @nickhale produces:
```
[x, w] = jacpts(5, 1,1);
>> sum(w)

ans =

   1.333333333333333
```

I think we could be more fancy using the 'ratio of gamma' for 'asy' method (because the weights are computed independently, and because the accuracy is slightly better), and use 'beta/sum' for 'rec' method.

**Comment by nickhale** [comment]
I suggest we use the `gammaln()` approach when `n` is small, and the `ratioOfGammaFunction()`  approach when `n` is large. 

As a reminder, we should add a comment that an alternative approach would be to use `C = 2^(a+b+1)*beta(a+1, b+1)/sum(w)` but we don't do this as we prefer to compute the weights independently.

**Comment by aaustin141** [comment]
Can this be closed, now that I've merged #1711?

**Comment by LourencoPeixoto** [comment]
Yes, I think this can be closed. However we don't use the above @nickhale's suggestion yet. 

Could @aaustin141 reopen #1711 in order to attend this suggestion?

**Comment by aaustin141** [comment]
@LourencoPeixoto:  At this point, I think it's best to make additional changes here as a new pull request.  I'll leave this issue open until @nickhale's suggestion is sorted out.

**Comment by LourencoPeixoto** [comment]
Ok, @aaustin141.

**Comment by aaustin141** [comment]
Now that we've merged #1747 as well, can this issue be closed?

**Comment by LourencoPeixoto** [comment]
Yes, I think this issue can be closed now.
